File: Favorite/LC/3243.cpp
#include<iostream>
#include<vector>

using namespace std;

vector<int> shortestDistanceAfterQueries(int n, vector<vector<int>>& queries) {
    vector<int> dp(n);
    vector<vector<int>> prev(n);
    for(int i = 1; i < n; i++) {
        dp[i] = i;
        prev[i].push_back(i - 1);
    }
    vector<int> ans(queries.size());
    int k = 0;
    for(auto & med : queries) {
        prev[med[1]].push_back(med[0]);
        for(int i = med[1]; i < n; i++) {
            for(auto & j : prev[i]) {
                dp[i] = min(dp[i],dp[j] + 1);
            }
        }
        ans[k++] = dp[n - 1];
    }
    return ans;
}

int main() {
    int n = 5;
    vector<vector<int>> queries = {{0,4},{4,2},{2,3},{4,1}};
    vector<int> ans = shortestDistanceAfterQueries(n,queries);
    for(auto & i : ans) {
        cout << i << " ";
    }
    return 0;
}

File: Favorite/LC/3243.go
func shortestDistanceAfterQueries(n int, queries [][]int) []int {
    prev := make([][]int, n)
    dp := make([]int, n)
    for i := 1; i < n; i++ {
        prev[i] = append(prev[i], i - 1)
        dp[i] = i
    }
    var ans []int
    for _, med := range queries {
        prev[med[1]] = append(prev[med[1]], med[0])
        for i := med[1]; i < n; i++ {
            for _, j := range prev[i] {
                dp[i] = min(dp[i], dp[j] + 1)
            }
        }
        ans = append(ans,dp[n - 1])
    }
    return ans
}